File: src/test/java/com/koddy/server/common/mock/stub/StubOAuthLoginProcessor.java
package com.koddy.server.common.mock.stub;

import com.koddy.server.auth.application.adapter.OAuthLoginProcessor;
import com.koddy.server.auth.domain.model.oauth.OAuthProvider;
import com.koddy.server.auth.domain.model.oauth.OAuthTokenResponse;
import com.koddy.server.auth.domain.model.oauth.OAuthUserResponse;
import com.koddy.server.common.fixture.OAuthFixture;

public class StubOAuthLoginProcessor implements OAuthLoginProcessor {
    @Override
    public OAuthUserResponse login(final OAuthProvider provider, final String code, final String redirectUri, final String state) {
        final OAuthTokenResponse token = OAuthFixture.parseOAuthTokenByCode(code);
        return OAuthFixture.parseOAuthUserByAccessToken(token.accessToken());
    }
}

File: src/test/java/com/koddy/server/common/mock/stub/StubOAuthLoginProcessor.kt
package com.koddy.server.common.mock.stub

import com.koddy.server.auth.application.adapter.OAuthLoginProcessor
import com.koddy.server.auth.domain.model.oauth.OAuthProvider
import com.koddy.server.auth.domain.model.oauth.OAuthUserResponse
import com.koddy.server.common.fixture.MenteeFixtureStore.menteeFixture
import com.koddy.server.common.fixture.MentorFixtureStore.mentorFixture
import com.koddy.server.common.utils.OAuthDummy.parseAuthorizationCode

open class StubOAuthLoginProcessor : OAuthLoginProcessor {
    override fun login(
        provider: OAuthProvider,
        code: String,
        redirectUri: String,
        state: String,
    ): OAuthUserResponse {
        val id: Long = parseAuthorizationCode(code)
        return when {
            code.contains("Mentor") -> mentorFixture(id = id).toGoogleUserResponse()
            else -> menteeFixture(id = id).toGoogleUserResponse()
        }
    }
}
